BxlAsLib.txt

```diff
@@ -0,0 +1,377 @@
+BuildXL as a library - aka. BXLLib.
+
+Problem statement
+-----------------
+- Performance
+  - Graph construction takes a long time due to DScript parsing, type checking, and evaluation.
+  - DScript frontend is not scalable to handle Office multi-architecture (mularchy) builds.
+
+- Frontend usability
+  - CMake, MsBuild frontends create their own adhoc JSON formats of build graphs.
+
+- Binary graph fragments are proposed as a way to construct build graphs, but need a way (API)
+  to represent and construct those fragments.
+
+Requirements
+------------
+- API must preserve compatibility, i.e., maintain backward compatibility.
+  - API must be binary compatible.
+  - Graph fragment must be versioned.

```

Comment:
Given your comment about 'frontend usabillity' , the binary format should be language independent so build engines implemented in non .net languages like java or c++ can participate